frontend/expr_inferrer.py

```diff
@@ -375,33 +376,69 @@ def _get_args_types(args, context, instance, solver):
     return args_types
 
 
-def _infer_annotated_function_call(args_types, solver, annotations):
-    result_type = solver.new_z3_const("call")
-    solver.annotation_resolver.add_annotated_function_axioms(args_types, solver, annotations, result_type)
+def _infer_annotated_function_call(args_types, solver, annotations, result_type):
+    return solver.annotation_resolver.get_annotated_function_axioms(args_types, solver, annotations, result_type)
 
-    return result_type
+
+def _get_builtin_method_call_axioms(args_types, solver, context, result_type, method_name):
+    """Get the axioms of built-in method calls"""
+    possible_methods = context.get_matching_methods(method_name)
+    method_axioms = []
+    for method in possible_methods:
+        cur_method_axioms = _infer_annotated_function_call(args_types, solver, method, result_type)
+        if cur_method_axioms is not None:
+            method_axioms.append(cur_method_axioms)
+    return method_axioms
 
 
 def infer_func_call(node, context, solver):
```

Comment:
The purpose of `instance` here is to act as `receiver` in case of class methods calls. So initially it is `None`, because there's no receiver for non-method function calls. And whenever we have attribute-function call, the instance will have the type of this receiver and it will be used as the first argument in the function call. However, there is a case where `instance` may be an imported `module` type, and in such case, we don't want it to act as a receiver, that's why it's reset to `None`.

So more formally:
```
If the call is method call:
    if module access:
        no receiver
    else:
        add receiver instance
else:
    no receiver
```

I'll try to change the way it's written to make to more clear.